storage_scrubber/src/checks.rs

```diff
@@ -165,16 +165,17 @@ pub(crate) async fn branch_cleanup_and_check_errors(
                                 .head_object(&path, &CancellationToken::new())
                                 .await;
 
-                            if response.is_err() {
```

Comment:
I mean, the original code here is clearly not correct. It will fail even on e.g. rate limiting errors or transient 5xx errors. We need to check for 404 (or whatever code is returned for a missing object), anything else should be retried or logged as an error.

I'm fine with adding logging, but we may as well just fix the code to check for 404 errors.

Reply:
We can get some signals from staging and if I'm confident enough I'll just make it only check 404 before the next release.